ImageData/Hdf5Loader.h

```diff
@@ -293,13 +296,138 @@ bool Hdf5Loader::GetCursorSpectralData(std::vector<float>& data, int stokes, int
+
+        auto& num_pixels = stats[CARTA::StatsType::NumPixels];
+        auto& nan_count = stats[CARTA::StatsType::NanCount];
+        auto& sum = stats[CARTA::StatsType::Sum];
+        auto& mean = stats[CARTA::StatsType::Mean];
+        auto& rms = stats[CARTA::StatsType::RMS];
+        auto& sigma = stats[CARTA::StatsType::Sigma];
+        auto& sum_sq = stats[CARTA::StatsType::SumSq];
+        auto& min = stats[CARTA::StatsType::Min];
+        auto& max = stats[CARTA::StatsType::Max];
+
+        std::vector<float> slice_data(num_z * num_y);
+
+        // Set initial values of stats which will be incremented (we may have expired region data)
+        for (size_t z = 0; z < num_z; z++) {
+            min[z] = FLT_MAX;
+            max[z] = FLT_MIN;
+            num_pixels[z] = 0;
+            nan_count[z] = 0;
+            sum[z] = 0;
+            sum_sq[z] = 0;
+        }
+
+        // Load each X slice of the swizzled region bounding box and update Z stats incrementally
+        for (size_t x = 0; x < num_x; x++) {
+            casacore::Slicer slicer;
+            if (_num_dims == 4) {
+                slicer = casacore::Slicer(IPos(4, 0, y_min, x + x_min, stokes), IPos(4, num_z, num_y, 1, 1));
+            } else if (_num_dims == 3) {
+                slicer = casacore::Slicer(IPos(3, 0, y_min, x + x_min), IPos(3, num_z, num_y, 1));
+            }
+
+            casacore::Array<float> tmp(slicer.length(), slice_data.data(), casacore::StorageInitPolicy::SHARE);
+
+            try {
+                LoadSwizzledData(FileInfo::Data::SWIZZLED)->doGetSlice(tmp, slicer);
+            } catch (casacore::AipsError& err) {
+                std::cerr << "Could not load cursor spectral data from swizzled HDF5 dataset. AIPS ERROR: " << err.getMesg() << std::endl;
+                return nullptr;
+            }
+
+            for (size_t y = 0; y < num_y; y++) {
+                // skip all Z values for masked pixels
+                if (!mask->getAt(IPos(2, x, y))) {
+                    continue;
+                }
+                for (size_t z = 0; z < num_z; z++) {
+                    float& v = slice_data[y * num_z + z];
+
+                    if (std::isfinite(v)) {
+                        num_pixels[z] += 1;
+
+                        sum[z] += v;
+                        sum_sq[z] += v * v;
+
+                        if (v < min[z]) {
+                            min[z] = v;
+                        } else if (v > max[z]) {
+                            max[z] = v;
+                        }
+
+                    } else {
+                        nan_count[z] += 1;
+                    }
+                }
+            }
+        }
+
+        float mean_sq;
+
+        // Calculate final stats
+        for (size_t z = 0; z < num_z; z++) {
+            if (num_pixels[z]) {
+                mean[z] = sum[z] / num_pixels[z];
+
+                mean_sq = sum_sq[z] / num_pixels[z];
+                rms[z] = sqrtf(mean_sq);
```

Comment:
stats are all done using `double` precision, so this should be `sqrt` in order to give the same results as casacore